snorkel/labeling/utils.py

```diff
@@ -27,3 +28,45 @@ def filter_unlabeled_dataframe(
     """
     mask = (L != -1).any(axis=1)
     return X.iloc[mask], y[mask]
+
+
+def create_cv_data(
+    L_train: np.ndarray, L_dev: np.ndarray, Y_dev: np.ndarray
+) -> PredefinedSplit:
+    """
+    Create predefined cross validation split and corresponding L and Y matrix with train and dev splits.
+
+    Returns combined L, Y matrix with corresponding CV split object with two splits.
+
+    Parameters
+    ----------
+    L_train
+        An [n,m] matrix with values in {-1,0,1,...,k-1} used for training
+    L_dev
+        An [n,m] matrix with values in {-1,o0,1,...,k-1} used for scoring
+    Y_dev
+        Gold labels associated with L_dev
+
+    Returns
+    -------
+    L
+        Combined L_train and L_dev matrix
+    Y
+        Combined Y_train (all -1s) and Y_dev array
+    cv_split
+        PredefinedSplit object with two splits: L_train in train and *_dev in test
+    """
+
+    if np.shape(L_dev)[0] != np.shape(Y_dev)[0]:
+        raise ValueError("Num. datapoints in Y_dev and L_dev do not match.")
+
+    # combine train and dev L and Y
+    L_all = np.append(L_train, L_dev, axis=0)
+    Y_all = np.append(-1 * np.ones(np.shape(L_train)[0]), Y_dev)
+
+    # create cv split array with two splits
+    test_fold = np.append(
+        -1 * np.ones(np.shape(L_train)[0]), np.ones(np.shape(L_dev)[0])
```

Comment:
let's do `n_train = np.shape(L_train)[0]` and `n_dev = np.shape(L_dev)[0]` at the beginning since we use so much